packages/babel-parser/src/plugins/flow/index.js

```diff
@@ -2837,17 +2838,12 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         state = this.state.clone();
 
         jsx = this.tryParse(
-          () =>
-            super.parseMaybeAssign(
-              refExpressionErrors,
-              afterLeftParse,
-              refNeedsArrowPos,
-            ),
+          () => super.parseMaybeAssign(refExpressionErrors, afterLeftParse),
           state,
         );
         /*:: invariant(!jsx.aborted) */
 
-        if (!jsx.error) return jsx.node;
+        if (!jsx.error && jsx.node) return jsx.node;
```

Comment:
Q: why should we check `jsx.node`?

Reply:
When I change `SyntaxError` to `ParsingError` [there](https://github.com/babel/babel/pull/13419#discussion_r650199238)

It pop me a flow error. And as you saw I rarely (and it's probably not good) use :
```flow
/*:: invariant(jsx.node != null) */
```flow

at the end of the day I see two solution:

- Let it like this
- use invariant comment

What do you prefer ?

Reply:
Since when `jsx.error` is not defined here, `jsx.node` is always defined, it's ok to use an invariant comment to let the compiler know what we know.